lib/matplotlib/legend_handler.py

```diff
@@ -734,32 +733,30 @@ class HandlerPolyCollection(HandlerBase):
     """
     def _update_prop(self, legend_handle, orig_handle):
         def first_color(colors):
-            if colors is None:
-                return None
-            colors = mcolors.to_rgba_array(colors)
-            if len(colors):
-                return colors[0]
-            else:
-                return "none"
+            if colors.size == 0:
+                return (0, 0, 0, 0)
+            return tuple(colors[0])
 
         def get_first(prop_array):
             if len(prop_array):
                 return prop_array[0]
             else:
                 return None
-        edgecolor = getattr(orig_handle, '_original_edgecolor',
-                            orig_handle.get_edgecolor())
-        legend_handle.set_edgecolor(first_color(edgecolor))
-        facecolor = getattr(orig_handle, '_original_facecolor',
-                            orig_handle.get_facecolor())
-        legend_handle.set_facecolor(first_color(facecolor))
-        legend_handle.set_fill(orig_handle.get_fill())
-        legend_handle.set_hatch(orig_handle.get_hatch())
+
+        # orig_handle is a PolyCollection and legend_handle is a Patch.
+        # Directly set Patch color attributes (must be RGBA tuples).
+        legend_handle._facecolor = first_color(orig_handle.get_facecolor())
```

Comment:
This is more invasive than my change so a couple of questions.  1) why set directly?  2) does this cover more cases, in which case should we add more tests?

Reply:
1. We want all the special-case handling to be done by the PolyCollection, after which we simply hand off the results to the Patch.  See https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/patches.py#L249 for another example of this approach.
2. I don't know.  I think it is more a question of whether enough legend with/without hatching cases are tested than of any expected changes in behavior.
More invasive? You could describe it that way, but I view it as a cleanup and an overall simplification.